presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/BaseArrowFlightClientHandler.java

```diff
@@ -0,0 +1,151 @@
+import static com.facebook.plugin.arrow.ArrowErrorCode.ARROW_FLIGHT_INFO_ERROR;
+import static com.facebook.plugin.arrow.ArrowErrorCode.ARROW_FLIGHT_METADATA_ERROR;
+import static java.nio.file.Files.newInputStream;
+import static java.util.Objects.requireNonNull;
+
+public abstract class BaseArrowFlightClientHandler
+{
+    private final ArrowFlightConfig config;
+    private final BufferAllocator allocator;
+
+    public BaseArrowFlightClientHandler(BufferAllocator allocator, ArrowFlightConfig config)
+    {
+        this.allocator = requireNonNull(allocator, "allocator is null");
+        this.config = requireNonNull(config, "config is null");
+    }
+
+    protected FlightClient createFlightClient()
+    {
+        Location location;
+        if (config.getArrowFlightServerSslEnabled() != null && !config.getArrowFlightServerSslEnabled()) {
+            location = Location.forGrpcInsecure(config.getFlightServerName(), config.getArrowFlightPort());
+        }
+        else {
+            location = Location.forGrpcTls(config.getFlightServerName(), config.getArrowFlightPort());
+        }
+        return createFlightClient(location);
+    }
+
+    protected FlightClient createFlightClient(Location location)
+    {
+        try {
+            Optional<InputStream> trustedCertificate = Optional.empty();
+            FlightClient.Builder flightClientBuilder = FlightClient.builder(allocator, location);
+            if (config.getVerifyServer() != null && !config.getVerifyServer()) {
+                flightClientBuilder.verifyServer(false);
+            }
+            else if (config.getFlightServerSSLCertificate() != null) {
+                trustedCertificate = Optional.of(newInputStream(Paths.get(config.getFlightServerSSLCertificate())));
+                flightClientBuilder.trustedCertificates(trustedCertificate.get()).useTls();
+            }
+
+            FlightClient flightClient = flightClientBuilder.build();
+            if (trustedCertificate.isPresent()) {
+                trustedCertificate.get().close();
+            }
+
+            return flightClient;
+        }
+        catch (Exception e) {
+            throw new ArrowException(ARROW_FLIGHT_CLIENT_ERROR, "Error creating flight client: " + e.getMessage(), e);
+        }
+    }
+
+    public abstract CallOption[] getCallOptions(ConnectorSession connectorSession);
+
+    protected FlightInfo getFlightInfo(FlightDescriptor flightDescriptor, ConnectorSession connectorSession)
+    {
+        try (FlightClient client = createFlightClient()) {
+            CallOption[] callOptions = getCallOptions(connectorSession);
+            return client.getInfo(flightDescriptor, callOptions);
+        }
+        catch (InterruptedException e) {
+            throw new ArrowException(ARROW_FLIGHT_INFO_ERROR, "Error getting flight information: " + e.getMessage(), e);
+        }
+    }
+
+    protected ClientClosingFlightStream getFlightStream(ArrowSplit split, ConnectorSession connectorSession)
```

Comment:
Removed the Ticket parameter. Since it is in the ArrowSplit already, we can just deserialize once